# Развитие игровой индустрии с 2000 по 2013 год.

- Автор: Никоноров Алексей
- Дата: 16.06.2026

### Цели и задачи проекта

В этой тетрадке используется датасет `/datasets/new_games.csv`, который содержит информацию о развитие игровой индустрии с 2000 по 2013 год. Будет проведено закомство с данными, проверка их корректности и предобработка, вывод необходимого среза данных.

### Описание данных

Данные `/datasets/new_games.csv` содержат информацию о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр:
- `Name` — название игры.
- `Platform` — название платформы.
- `Year of Release` — год выпуска игры.
- `Genre` — жанр игры.
- `NA sales` — продажи в Северной Америке (в миллионах проданных копий).
- `EU sales` — продажи в Европе (в миллионах проданных копий).
- `JP sales` — продажи в Японии (в миллионах проданных копий).
- `Other sales` — продажи в других странах (в миллионах проданных копий).
- `Critic Score` — оценка критиков (от 0 до 100).
- `User Score` — оценка пользователей (от 0 до 10).
- `Rating` — рейтинг организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.

### Содержимое проекта

1. [Загрузка данных и знакомство с ними](#1razd)
2. [Проверка ошибок в данных и их предобработка](#2razd)
3. [Фильтрация данных](#3razd)
4. [Категоризация данных](#4razd)
5. [Итоговый вывод](#5razd)

<a id='1razd'></a>
## 1. Загрузка данных и знакомство с ними 

- выгрузка данных из датасета `/datasets/new_games.csv` и знакомство с ними.


In [1]:
# импорт библиотеки pandas
import pandas as pd

In [2]:
# Выгрузка данных из датасета /datasets/new_games.csv в датафрейм game
game = pd.read_csv('/datasets/new_games.csv')

In [3]:
# Вывод информации о датафрейме
game.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


In [4]:
# доля пропусков в столбцах
game.isna().sum() / game.shape[0]

Name               0.000118
Platform           0.000000
Year of Release    0.016218
Genre              0.000118
NA sales           0.000000
EU sales           0.000000
JP sales           0.000000
Other sales        0.000000
Critic Score       0.513918
User Score         0.401274
Rating             0.405225
dtype: float64

In [5]:
# Вывод первых строк датафрейма
game.head()

,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN


Полученные данные представляют собой таблицу с 16 956 строками и 11 столбцами, содержащую информацию из датасета `/datasets/new_games.csv` об игровых релизах.

Данные соответствуют стандартному датасету по видеоиграм и включают информацию о названиях игр, игровых платформах, годах выпуска, жанрах, объёме продаж в разных регионах — Северной Америке, Европе, Японии и других странах, а также содержат оценки критиков и пользователей и указание возрастного рейтинга.
 
Данные в 6 столбцах `Name`, `Year of Releese`, `Genre`, `Critic Score`, `User Score`, `Rating` содержат много пропусков, особенно в столбцах с оценками. Это необходимо учесть при анализе.

7 столбцов имеют типы данных `object` и 4 числовые значения с плавающей точкой `float64`. Важно проверить столбцы `EU sales`,  `JP sales` и `User Score` — они выглядят как числа, но объявлены как `object`, возможно, в них есть нечисловые значения, мешающие преобразованию.

Текущие названия столбцов содержат пробелы и смешанный регистр, что мешает удобной работе в Python

 <a id='2razd'></a>

## 2.  Проверка ошибок в данных и их предобработка


### 2.1. Названия, или метки, столбцов датафрейма

- Выводим на экран названия всех столбцов датафрейма.
- Приведем все столбцы к стилю snake case. Названия должны быть в нижнем регистре, а вместо пробелов — подчёркивания.

In [6]:
# Вывод названий столбцов
game.columns

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='object')

In [7]:
# перевод названий столбцов к стилю snake case и к нижнему регистру
game.columns = game.columns.str.lower().str.replace(' ', '_')

### 2.2. Типы данных



Поле `year_of_release` содержит год выпуска игры и пропуски в данном поле могут свидетельствовать об отсутствии точной информации о дате выхода игры.
Поскольку доля пропусков в столбце `year_of_release` составляет около 1.6% существенного влияния на анализ это не окажет, пропуски в данном поле заменим на 0, чтобы привести к типу данных integer

In [8]:
# заменим пропуски в поле year_of_release на 0
game['year_of_release'] = game['year_of_release'].fillna(0)

In [9]:
# приведет поле year_of_release к типу int с понижением разрядности
game['year_of_release'] = pd.to_numeric(game['year_of_release'], downcast='integer')

В числовых столбцах могут встретиться строковые значения, например `unknown` или другие. Приведем такие столбцы к числовому типу данных, заменив строковые значения на пропуски.

In [10]:
game['eu_sales'] = pd.to_numeric(game['eu_sales'], errors='coerce')

In [11]:
game['jp_sales'] = pd.to_numeric(game['jp_sales'], errors='coerce')

In [12]:
# выводим уникальные значения поля `user_score`
game['user_score'].unique()

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

В столбце `user_score` с пользовательскими оценками помимо числовых значений содержатся пропуски - NaN и значение 'tbd', что вероятно расшифровывается как 'to be determined', т.е. оценки не определены. Заменим эти значения на NaN и преведем к типу данных `float`.

In [13]:
game['user_score'] = pd.to_numeric(game['user_score'], errors='coerce')

In [14]:
# выведем информацию о датафрейме после преобразования столбцов
game.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16954 non-null  object 
 1   platform         16956 non-null  object 
 2   year_of_release  16956 non-null  int16  
 3   genre            16954 non-null  object 
 4   na_sales         16956 non-null  float64
 5   eu_sales         16950 non-null  float64
 6   jp_sales         16952 non-null  float64
 7   other_sales      16956 non-null  float64
 8   critic_score     8242 non-null   float64
 9   user_score       7688 non-null   float64
 10  rating           10085 non-null  object 
dtypes: float64(6), int16(1), object(4)
memory usage: 1.3+ MB


### 2.3. Наличие пропусков в данных

- Считаем количество пропусков в каждом столбце в абсолютных и относительных значениях.


In [15]:
# количество пропусков в абсолютном значении
game.isna().sum()

name                  2
platform              0
year_of_release       0
genre                 2
na_sales              0
eu_sales              6
jp_sales              4
other_sales           0
critic_score       8714
user_score         9268
rating             6871
dtype: int64

In [16]:
# количество пропусков в относительном значении
game.isna().sum() / game.shape[0]

name               0.000118
platform           0.000000
year_of_release    0.000000
genre              0.000118
na_sales           0.000000
eu_sales           0.000354
jp_sales           0.000236
other_sales        0.000000
critic_score       0.513918
user_score         0.546591
rating             0.405225
dtype: float64

После преобразования типа данных, пропуски содержаться в 7 столбцах.
- `rating` - 6871 пропуск, около 40%. Пропуски вероятно вызваны отсутствием данных, т.к. ESRB - рейтинг для игр, выпущенных на территории США и Канады, для других регионов данный рейтинг может отсутствовать и иметь другую маркировку. Всвязи с этим целесообразно заменить пропущенные значения на 'unknown'.
- `name`, `genre` - по 2 пропуска. Возможные причины пропусков - ошибка ввода или повреждение данных. Их доля составляет 0.01%, что не окажет существенного влияния на анализ данных и строки с пропусками можно удалить.
- `eu_sales` - 6 пропусков и `jp_sales` - 4 пропуска, что составляет 0.03% и 0.02% соответственно. Причина возникновения может быть связана с отсутствием данных о продажах в регионе. Количество пропусков не окажет существенного влияния на анализ данных и строки с пропусками можно удалить.
- `critic_score` и `user_score` содержат 8714 и 9268 пропусков в строках, что составляет 51% и 54% соответственно. Это значительное количество пропусков. Скорее всего это пропуски MNAR — мы не знаем были проведены оценки этих игр или нет. Заменим пропуски уникальным значением(-1)

In [17]:
game['critic_score'] = game['critic_score'].fillna(-1)
game['user_score'] = game['user_score'].fillna(-1)

In [18]:
# заменим пропущенные значения на 'unknown'
game['rating'] = game['rating'].fillna('unknown')

In [19]:
# Находим все строки, где name, genre, eu_sales, jp_sales имеют пропуска
clean_games = game[game['name'].isnull() | game['genre'].isnull() | game['eu_sales'].isnull() | game['jp_sales'].isnull()]
display(clean_games)

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating
446,Rhythm Heaven,DS,2008,Misc,0.55,NaN,1.93,0.13,83.0,9.0,E
467,Saints Row 2,X360,2008,Action,1.94,0.79,NaN,0.28,81.0,8.1,M
661,NaN,GEN,1993,NaN,1.78,0.53,0.00,0.08,-1.0,-1.0,unknown
802,Dead Rising,X360,2006,Action,1.16,NaN,0.08,0.20,85.0,7.6,M
819,UFC 2009 Undisputed,X360,2009,Fighting,1.48,0.39,NaN,0.19,83.0,7.9,T
1131,Prince of Persia: Warrior Within,PS2,2004,Action,0.54,NaN,0.00,0.22,83.0,8.5,M
1132,Far Cry 4,XOne,2014,Shooter,0.80,NaN,0.01,0.14,82.0,7.5,M
1379,Hello Kitty Party,DS,2007,Misc,0.78,0.51,NaN,0.12,-1.0,-1.0,E
1394,Sonic Advance 3,GBA,2004,Platform,0.74,NaN,0.08,0.06,79.0,8.4,E
1612,Ratatouille,DS,2007,Action,0.49,NaN,0.00,0.14,-1.0,-1.0,unknown


In [20]:
# удаляем строки с пропусками в столбцах name, genre и создаем копию датафрейма gamedrop, с которой продолжим работу
gamedrop = game.dropna(subset=['name', 'genre', 'eu_sales', 'jp_sales']).copy()

In [21]:
# выводим информацию о датафрейме после обработки пропусков
gamedrop.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16944 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16944 non-null  object 
 1   platform         16944 non-null  object 
 2   year_of_release  16944 non-null  int16  
 3   genre            16944 non-null  object 
 4   na_sales         16944 non-null  float64
 5   eu_sales         16944 non-null  float64
 6   jp_sales         16944 non-null  float64
 7   other_sales      16944 non-null  float64
 8   critic_score     16944 non-null  float64
 9   user_score       16944 non-null  float64
 10  rating           16944 non-null  object 
dtypes: float64(6), int16(1), object(4)
memory usage: 1.5+ MB


### 2.4. Явные и неявные дубликаты в данных



- Изучим столбцы `platform`, `year_of_release`, `genre`, `rating`.
- Проведем нормализацию данных с текстовыми значениями. Названия и жанры игр приведем к нижнему регистру, а названия рейтинга — к верхнему и уберем лишние пробелы в названиях . После чего выведем количество уникальных строк с изменениями и до.

In [22]:
# Список столбцов для проверки
columns_to_check = ['platform', 'year_of_release', 'genre', 'rating']

for column in columns_to_check:
    print(f'Уникальные значения в столбце {column}:')
    
    unique_sorted = sorted(gamedrop[column].sort_values().unique())
    print(unique_sorted)
    print() 

Уникальные значения в столбце platform:
['2600', '3DO', '3DS', 'DC', 'DS', 'GB', 'GBA', 'GC', 'GEN', 'GG', 'N64', 'NES', 'NG', 'PC', 'PCFX', 'PS', 'PS2', 'PS3', 'PS4', 'PSP', 'PSV', 'SAT', 'SCD', 'SNES', 'TG16', 'WS', 'Wii', 'WiiU', 'X360', 'XB', 'XOne']

Уникальные значения в столбце year_of_release:
[0, 1980, 1981, 1982, 1983, 1984, 1985, 1986, 1987, 1988, 1989, 1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016]

Уникальные значения в столбце genre:
['ACTION', 'ADVENTURE', 'Action', 'Adventure', 'FIGHTING', 'Fighting', 'MISC', 'Misc', 'PLATFORM', 'PUZZLE', 'Platform', 'Puzzle', 'RACING', 'ROLE-PLAYING', 'Racing', 'Role-Playing', 'SHOOTER', 'SIMULATION', 'SPORTS', 'STRATEGY', 'Shooter', 'Simulation', 'Sports', 'Strategy']

Уникальные значения в столбце rating:
['AO', 'E', 'E10+', 'EC', 'K-A', 'M', 'RP', 'T', 'unknown']



In [23]:
# Создадим новые нормализованные столбцы
gamedrop['genre'] = gamedrop['genre'].str.lower().str.strip()
gamedrop['rating'] = gamedrop['rating'].str.upper().str.strip()
gamedrop['name'] = gamedrop['name'].str.lower().str.strip()

In [24]:
display("Количество уникальных платформ (до):", game['genre'].nunique())

display("Количество уникальных платформ (после):", gamedrop['genre'].nunique())


'Количество уникальных платформ (до):'

24

'Количество уникальных платформ (после):'

12

In [25]:
display("Количество уникального рейтинга (до):", game['rating'].nunique())

display("Количество уникального рейтинга (после):", gamedrop['rating'].nunique())


'Количество уникального рейтинга (до):'

9

'Количество уникального рейтинга (после):'

9

In [26]:
display("Количество уникальных игр (до):", game['name'].nunique())

display("Количество уникальных игр (после):", gamedrop['name'].nunique())


'Количество уникальных игр (до):'

11559

'Количество уникальных игр (после):'

11556

- После нормализации данных и устранения неявных дубликатов, проверим наличие явных дубликатов в данных.

In [27]:
# считаем количество явных дубликатов
duplicates = gamedrop.duplicated(subset=['name', 'genre', 'platform', 'year_of_release', 'rating'], keep='first').sum()

In [28]:
display("явных дубликатов:", duplicates)

'явных дубликатов:'

242

In [29]:
# удаляем явные дубликаты и сохраняем новый датафрейм dfgame
dfgame = gamedrop.drop_duplicates(
    subset=['name', 'genre', 'platform', 'year_of_release', 'rating'],
    keep='first' 
).reset_index(drop=True)

In [30]:
# выводим информацию по датафрейму dfgame
dfgame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16702 entries, 0 to 16701
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16702 non-null  object 
 1   platform         16702 non-null  object 
 2   year_of_release  16702 non-null  int16  
 3   genre            16702 non-null  object 
 4   na_sales         16702 non-null  float64
 5   eu_sales         16702 non-null  float64
 6   jp_sales         16702 non-null  float64
 7   other_sales      16702 non-null  float64
 8   critic_score     16702 non-null  float64
 9   user_score       16702 non-null  float64
 10  rating           16702 non-null  object 
dtypes: float64(6), int16(1), object(4)
memory usage: 1.3+ MB


In [31]:
print(f"Размер исходного датафрейма: {len(game)}")
print(f"Размер очищенного датафрейма: {len(dfgame)}")
print(f"Удалено строк: {len(game) - len(dfgame)}")
print(f"Удалено строк в относительном значении: {round((len(game) - len(dfgame)) / len(game), 3)}")

Размер исходного датафрейма: 16956
Размер очищенного датафрейма: 16702
Удалено строк: 254
Удалено строк в относительном значении: 0.015


- В процессе подготовки данных датафрейм был изменен. Было удалено 254 строки(пропуски и дубликаты), что составляет около 1.5% данных.

- На этапе предобработки данных был проведён комплексный анализ и очистка датафрейма с информацией об играх, что позволило привести данные к единому и пригодному для дальнейшего анализа виду.
- Названия столбцов были приведены к нижнему регистру, пробелы в названиях заменены на `_`, для удобства анализа.
- Были изменены типы данных в столбцах `year_of_release`, `eu_sales`, `jp_sales`, `user_score`
- Были выявлены и удалены явные дубликаты — строки, полностью идентичные по ключевым полям: `name`, `genre`, `platform`, `year_of_release`,`rating`.
- Перед этим выполнена нормализация текстовых полей (`name`, `genre`, `rating`) — приведение к единому регистру, удаление лишних пробелов — что позволило обнаружить неявные дубликаты.
- Для числовых столбцов `critic_score` и `user_score` пропущенные значения были заменены на значение-индикатор(-1)
- Удалены пропуски в столбцах `name`, `genre`, `eu_sales`, `jp_sales`.


In [32]:
# вывод количества удаленных строк в абсолютном и относительном значении
print(f"Удалено строк: {len(game) - len(dfgame)}")
print(f"Удалено строк в относительном значении: {round((len(game) - len(dfgame)) / len(game) * 100.0, 2)}%")

Удалено строк: 254
Удалено строк в относительном значении: 1.5%


- В процессе предобработки данных было удалено 254 строки, что составляет 1.5% датафрейма.

<a id='3razd'></a>

## 3. Фильтрация данных 

Коллеги хотят изучить историю продаж игр в начале XXI века, и их интересует период с 2000 по 2013 год включительно. Отберем данные по этому показателю и сохраним новый срез данных в отдельном датафрейме `df_actual`.

In [33]:
# фильтруем данные и создаем новый датафрейм df_actual
df_actual = dfgame.copy()
df_actual = df_actual.loc[(df_actual['year_of_release'] >= 2000) & (df_actual['year_of_release'] <= 2013)]

In [34]:
# выводим информацию по отфильтрованным данным
df_actual.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12771 entries, 0 to 16700
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             12771 non-null  object 
 1   platform         12771 non-null  object 
 2   year_of_release  12771 non-null  int16  
 3   genre            12771 non-null  object 
 4   na_sales         12771 non-null  float64
 5   eu_sales         12771 non-null  float64
 6   jp_sales         12771 non-null  float64
 7   other_sales      12771 non-null  float64
 8   critic_score     12771 non-null  float64
 9   user_score       12771 non-null  float64
 10  rating           12771 non-null  object 
dtypes: float64(6), int16(1), object(4)
memory usage: 1.1+ MB


In [35]:
# выводим первые строки отфильтрованной таблицы df_actual
df_actual.head()

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating
0,wii sports,Wii,2006,sports,41.36,28.96,3.77,8.45,76.0,8.0,E
2,mario kart wii,Wii,2008,racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,wii sports resort,Wii,2009,sports,15.61,10.93,3.28,2.95,80.0,8.0,E
6,new super mario bros.,DS,2006,platform,11.28,9.14,6.50,2.88,89.0,8.5,E
7,wii play,Wii,2006,misc,13.96,9.18,2.93,2.84,58.0,6.6,E


- Новый, отфильтрованный по годам датафрейм, содержит 12771 строку и 11 столбцов. 6 столбцов типа `float64`, 1 типа `int16` и 4 типа `object`

<a id='4razd'></a>

## 4. Категоризация данных 
    
Проведем категоризацию данных:
- Разделим все игры по оценкам пользователей и выделим такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

In [36]:
# создаем столбец users_category и отражаем в нем разделение оценок пользователей на категории 
df_actual['users_category'] = pd.cut(
    df_actual['user_score'],
    bins=[0, 3, 8, 10],
    labels=['низкая оценка', 'средняя оценка', 'высокая оценка'],
    right=False 
)

- Разделим все игры по оценкам критиков и выделим такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [37]:
# создаем столбец critics_category и отражаем в нем разделение оценок пользователей на категории
df_actual['critics_category'] = pd.cut(
    df_actual['critic_score'],
    bins=[0, 30, 80, 100],
    labels=['низкая оценка', 'средняя оценка', 'высокая оценка'],
    right=False 
)

In [38]:
# сбросим индексы и выведем первые строки измененной таблицы df_actual с новыми столбцами users_category и critics_category
df_actual = df_actual.reset_index(drop=True)
df_actual.head()

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating,users_category,critics_category
0,wii sports,Wii,2006,sports,41.36,28.96,3.77,8.45,76.0,8.0,E,высокая оценка,средняя оценка
1,mario kart wii,Wii,2008,racing,15.68,12.76,3.79,3.29,82.0,8.3,E,высокая оценка,высокая оценка
2,wii sports resort,Wii,2009,sports,15.61,10.93,3.28,2.95,80.0,8.0,E,высокая оценка,высокая оценка
3,new super mario bros.,DS,2006,platform,11.28,9.14,6.50,2.88,89.0,8.5,E,высокая оценка,высокая оценка
4,wii play,Wii,2006,misc,13.96,9.18,2.93,2.84,58.0,6.6,E,средняя оценка,средняя оценка


- После категоризации данных проверим результат: сгруппируем данные по выделенным категориям и посчитаем количество игр в каждой категории.

In [39]:
display("оценки пользователей", df_actual['users_category'].value_counts())

'оценки пользователей'

средняя оценка    4077
высокая оценка    2282
низкая оценка      116
Name: users_category, dtype: int64

In [40]:
display("оценки критиков", df_actual['critics_category'].value_counts())

'оценки критиков'

средняя оценка    5421
высокая оценка    1685
низкая оценка       55
Name: critics_category, dtype: int64

- Выделим топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [41]:
top_7 = df_actual['platform'].value_counts().head(7)

In [42]:
top_7

PS2     2126
DS      2117
Wii     1275
PSP     1179
X360    1118
PS3     1086
GBA      810
Name: platform, dtype: int64

<a id='5razd'></a>

## 5. Итоговый вывод 



#### На этапе предобработки данных был проведён комплексный анализ и очистка датафрейма с информацией об играх, что позволило привести данные к единому и пригодному для дальнейшего анализа виду.
- Названия столбцов были приведены к нижнему регистру, пробелы в названиях заменены на `_`, для удобства анализа.
- Были изменены типы данных в столбцах `year_of_release`, `eu_sales`, `jp_sales`, `user_score`
- Были выявлены и удалены явные дубликаты — строки, полностью идентичные по ключевым полям: `name`, `genre`, `platform`, `year_of_release`,`rating`.
- Перед этим выполнена нормализация текстовых полей (`name`, `genre`, `rating`) — приведение к единому регистру, удаление лишних пробелов — что позволило обнаружить неявные дубликаты.
- Для числовых столбцов `critic_score` и `user_score` пропущенные значения были заменены на значение-индикатор(-1)
- Удалены пропуски в столбцах `name`, `genre`, `eu_sales`, `jp_sales`.

#### В ходе анализа был выполнен комплексный этап подготовки и структурирования данных по видеоиграм с целью последующего углублённого исследования. Был создан актуальный срез данных и расширена структура датасета
- Исходный датасет был отфильтрован по временному периоду, релевантному для анализа:
Год выпуска игры: от 2000 до 2013 включительно.
Результат сохранён в датафрейме `df_actual`.
Этот срез позволяет сосредоточиться на современном этапе развития игровой индустрии, исключив устаревшие платформы.
- Сброшены индексы для корректной индексации.
- Проведена категоризация оценок пользователей и критиков. Добавлены новые столбцы `critics_category` и `users_category`.
- Выведен рейтинг топ-7 платформ по количеству игр, выпущенных за весь актуальный период.
